Support vNic type and Reservation affitinity

gogasca opened the pull request:
Now you can define the desired vNIC type and Reservation affinity features from Compute Engine when creating your Notebook instance.

<!-- Put a description of what this PR is for here, along with any references to issues that this resolves or contributes to -->

Support vNic type and Reservation affinity features from Compute Engine into Notebook instance creation request
API [reference](https://cloud.google.com/notebooks/docs/reference/rpc/google.cloud.notebooks.v1#google.cloud.notebooks.v1.Instance)

[vNIC type](https://cloud.google.com/compute/docs/networking/using-gvnic#create-instance-gvnic-image)
[Compute Reservations](https://cloud.google.com/compute/docs/instances/reserving-zonal-resources)

<!--
Replace each [ ] with [X] to check it. Switch to the preview view to make it easier to click on links.
These steps will speed up the review process, and we appreciate you spending time on them before sending
your code to be reviewed.
-->
If this PR is for Terraform, I acknowledge that I have:

- [X] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [X] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-downstream-tools), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [X] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/third_party/terraform/tests) (for handwritten resources or update tests).
- [X] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/master/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [X] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/master/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

<!-- AUTOCHANGELOG for Downstream PRs.

Please select one of the following "release-note:" headings:
    - release-note:enhancement
    
Unless you choose release-note:none, please add a release note.

See .ci/RELEASE_NOTES_GUIDE.md for writing good release notes.

You can add more release note blocks if you want more than one CHANGELOG
entry for this PR.
-->
**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
`notebooks`: added support for `nic_type`, `reservation_affinity` to `google_notebooks_instance`
```

modular-magician commented:
Hello!  I am a robot who works on Magic Modules PRs.

I have detected that you are a community contributor, so your PR will be assigned to someone with a commit-bit on this repo for initial review.

Thanks for your contribution!  A human will be with you soon.

@melinath, please review this PR or find an appropriate assignee.

melinath commented on mmv1/products/notebooks/api.yaml:
```diff
@@ -281,6 +281,42 @@ objects:
+          description: |
+            The type of vNIC to be used on this interface. This may be gVNIC or VirtioNet.
+      - !ruby/object:Api::Type::NestedObject
+      name: 'ReservationAffinity '
```
```suggestion
      name: 'ReservationAffinity'
```

melinath reviewed (changes requested):
Hi, @gogasca - thanks for your contribution! Please make sure the new fields you're adding are exercised by tests - by modifying an existing example (defined in [notebooks/terraform.yaml](https://github.com/GoogleCloudPlatform/magic-modules/blob/master/mmv1/products/notebooks/terraform.yaml)), adding a new example, or modifying the [handwritten test file](https://github.com/GoogleCloudPlatform/magic-modules/blob/master/mmv1/third_party/terraform/tests/resource_notebooks_instance_test.go.erb)

melinath reviewed (changes requested):
In addition to adding tests as described in the previous change request, it looks like your release notes are not properly formatted. Please review the release notes guide & update them. 

It also looks like the indentation in your yaml file is incorrect - please make sure that you're able to run the compiler locally as described in this repository's README.

modular-magician commented:
Hi!  I'm the modular magician.  Your PR generated some diffs in downstreams - here they are.
# Diff report:


Terraform GA: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-4981-old..auto-pr-4981) ( 2 files changed, 215 insertions(+))
Terraform Beta: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-4981-old..auto-pr-4981) ( 2 files changed, 215 insertions(+))
TF Conversion: [Diff](https://github.com/modular-magician/terraform-google-conversion/compare/auto-pr-4981-old..auto-pr-4981) ( 1 file changed, 80 insertions(+))

modular-magician commented:
I have triggered VCR tests in RECORDING mode for the following tests that failed during VCR: TestAccAssuredWorkloadsWorkload_basic|TestAccAssuredWorkloadsWorkload_full|TestAccBigqueryDataTransferConfig|TestAccBillingBudget_budgetFilterProjectsOrdering|TestAccBillingBudget_billingBudgetUpdate|TestAccCloudBuildTrigger_pubsub_config|TestAccCloudBuildTrigger_webhook_config|TestAccComposerEnvironment_withMaintenanceWindow|TestAccComputeBackendService_withBackend|TestAccComputeBackendService_withBackendAndMaxUtilization|TestAccComputeBackendService_withBackendAndIAP|TestAccComputeBackendService_withMaxConnectionsPerInstance|TestAccComputeBackendService_withMaxConnections|TestAccComputeBackendService_withMaxConnectionsPerEndpoint|TestAccComputeBackendService_withMaxRatePerEndpoint|TestAccComputeGlobalForwardingRule_globalForwardingRuleInternalExample|TestAccComputeGlobalForwardingRule_internalLoadBalancing|TestAccComputeHaVpnGateway_computeHaVpnGatewayEncryptedInterconnectExample|TestAccComputeInstanceGroup_rename|TestAccComputeRouter_computeRouterEncryptedInterconnectExample|TestAccContainerCluster_backend|TestAccContainerNodePool_withGPU|TestAccDataLossPreventionDeidentifyTemplate_dlpDeidentifyTemplateBasicExample|TestAccDataLossPreventionDeidentifyTemplate_dlpDeidentifyTemplateUpdate|TestAccDialogflowCXFlow_dialogflowcxFlowFullExample|TestAccDialogflowCXVersion_dialogflowcxVersionFullExample|TestAccDialogflowCXFlow_update|TestAccDialogflowCXVersion_update|TestAccNetworkServicesEdgeCacheKeyset_networkServicesEdgeCacheKeysetBasicExample|TestAccNetworkServicesEdgeCacheOrigin_networkServicesEdgeCacheOriginBasicExample|TestAccNetworkServicesEdgeCacheOrigin_networkServicesEdgeCacheOriginAdvancedExample|TestAccNetworkServicesEdgeCacheOrigin_updateAndImport|TestAccNetworkServicesEdgeCacheService_networkServicesEdgeCacheServiceBasicExample|TestAccNetworkServicesEdgeCacheService_networkServicesEdgeCacheServiceAdvancedExample|TestAccNetworkServicesEdgeCacheService_updateAndImport|TestAccPrivatecaCertificate_privatecaCertificateCsrExample|TestAccPrivatecaCertificateAuthority_privatecaCertificateAuthorityBasicExample|TestAccPrivatecaCertificate_privatecaCertificateConfigExample|TestAccPrivatecaCertificate_privatecaCertificateNoAuthorityExample|TestAccSecurityCenterNotificationConfig_sccNotificationConfigBasicExample|TestAccTags You can view the result here: https://ci-oss.hashicorp.engineering/viewQueued.html?itemId=197601

modular-magician commented:
Tests failed during RECORDING mode: TestAccDataLossPreventionDeidentifyTemplate_dlpDeidentifyTemplateBasicExample|TestAccNetworkServicesEdgeCacheOrigin_networkServicesEdgeCacheOriginAdvancedExample Please fix these to complete your PR

gogasca commented:
> We found a Contributor License Agreement for you (the sender of this pull request), but were unable to find agreements for all the commit author(s) or Co-authors. If you authored these, maybe you used a different email address in the git commits than was used to sign the CLA ([login here](https://cla.developers.google.com/) to double check)? If these were authored by someone else, then they will need to sign a CLA as well, and confirm that they're okay with these being contributed to Google.
> In order to pass this check, please resolve this problem and then comment `@googlebot I fixed it.`. If the bot doesn't comment, it means it doesn't think anything has changed.
> 
> ℹ️ **Googlers: [Go here](https://goto.google.com/prinfo/https%3A%2F%2Fgithub.com%2FGoogleCloudPlatform%2Fmagic-modules%2Fpull%2F4981) for more info**.

@googlebot I fixed it.

modular-magician commented:
Hi!  I'm the modular magician.  Your PR generated some diffs in downstreams - here they are.
# Diff report:


Terraform GA: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-4981-old..auto-pr-4981) ( 3 files changed, 231 insertions(+))
Terraform Beta: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-4981-old..auto-pr-4981) ( 3 files changed, 231 insertions(+))
TF Conversion: [Diff](https://github.com/modular-magician/terraform-google-conversion/compare/auto-pr-4981-old..auto-pr-4981) ( 1 file changed, 80 insertions(+))

gogasca commented:
> In addition to adding tests as described in the previous change request, it looks like your release notes are not properly formatted. Please review the release notes guide & update them.
> 
> It also looks like the indentation in your yaml file is incorrect - please make sure that you're able to run the compiler locally as described in [this repository's README](https://github.com/GoogleCloudPlatform/magic-modules#generating-the-terraform-providers).
> 
> Could you link to the documentation for this feature? I don't see it listed on https://cloud.google.com/notebooks/docs/reference/rest/v1beta1/projects.locations.instances#Instance.

With respect to beta, this feature is not in v1beta1 API, what's the best way to exclude it from beta?

modular-magician commented:
I have triggered VCR tests in RECORDING mode for the following tests that failed during VCR: TestAccAssuredWorkloadsWorkload_basic|TestAccAssuredWorkloadsWorkload_full|TestAccBigqueryDataTransferConfig|TestAccBigqueryReservationReservation_bigqueryReservationBasicExample|TestAccBillingBudget_budgetFilterProjectsOrdering|TestAccBillingBudget_billingBudgetUpdate|TestAccCloudBuildTrigger_pubsub_config|TestAccCloudBuildTrigger_webhook_config|TestAccComposerEnvironment_withMaintenanceWindow|TestAccComputeBackendService_withBackend|TestAccComputeBackendService_withBackendAndIAP|TestAccComputeBackendService_withBackendAndMaxUtilization|TestAccComputeBackendService_withMaxConnections|TestAccComputeBackendService_withMaxConnectionsPerInstance|TestAccComputeBackendService_withMaxConnectionsPerEndpoint|TestAccComputeBackendService_withMaxRatePerEndpoint|TestAccComputeGlobalForwardingRule_globalForwardingRuleInternalExample|TestAccComputeGlobalForwardingRule_internalLoadBalancing|TestAccComputeHaVpnGateway_computeHaVpnGatewayEncryptedInterconnectExample|TestAccComputeInstanceGroup_rename|TestAccComputeRouter_computeRouterEncryptedInterconnectExample|TestAccContainerCluster_backend|TestAccContainerNodePool_withGPU|TestAccDataLossPreventionDeidentifyTemplate_dlpDeidentifyTemplateBasicExample|TestAccDataLossPreventionDeidentifyTemplate_dlpDeidentifyTemplateUpdate|TestAccDialogflowCXFlow_dialogflowcxFlowFullExample|TestAccDialogflowCXVersion_dialogflowcxVersionFullExample|TestAccDialogflowCXFlow_update|TestAccDialogflowCXVersion_update|TestAccGKEHubMembership_gkehubMembershipBasicExample|TestAccGKEHubMembership_gkehubMembershipIssuerExample|TestAccNetworkServicesEdgeCacheKeyset_networkServicesEdgeCacheKeysetBasicExample|TestAccNetworkServicesEdgeCacheOrigin_networkServicesEdgeCacheOriginBasicExample|TestAccNetworkServicesEdgeCacheOrigin_networkServicesEdgeCacheOriginAdvancedExample|TestAccNetworkServicesEdgeCacheOrigin_updateAndImport|TestAccNetworkServicesEdgeCacheService_networkServicesEdgeCacheServiceBasicExample|TestAccNetworkServicesEdgeCacheService_networkServicesEdgeCacheServiceAdvancedExample|TestAccNetworkServicesEdgeCacheService_updateAndImport|TestAccNotebooksInstance_notebookInstanceFullExample|TestAccPrivatecaCertificate_privatecaCertificateCsrExample|TestAccPrivatecaCertificateAuthority_privatecaCertificateAuthorityBasicExample|TestAccPrivatecaCertificate_privatecaCertificateConfigExample|TestAccPrivatecaCertificate_privatecaCertificateNoAuthorityExample|TestAccSecurityCenterNotificationConfig_sccNotificationConfigBasicExample You can view the result here: https://ci-oss.hashicorp.engineering/viewQueued.html?itemId=197787

modular-magician commented:
Tests failed during RECORDING mode: TestAccDataLossPreventionDeidentifyTemplate_dlpDeidentifyTemplateBasicExample|TestAccNotebooksInstance_notebookInstanceFullExample|TestAccGKEHubMembership_gkehubMembershipBasicExample|TestAccPrivatecaCertificate_privatecaCertificateNoAuthorityExample|TestAccGKEHubMembership_gkehubMembershipIssuerExample Please fix these to complete your PR

gogasca commented:
Also quick question, not sure why cla is still set to No.
Thanks!

modular-magician commented:
Hi!  I'm the modular magician.  Your PR generated some diffs in downstreams - here they are.
# Diff report:


Terraform GA: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-4981-old..auto-pr-4981) ( 3 files changed, 217 insertions(+))
Terraform Beta: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-4981-old..auto-pr-4981) ( 3 files changed, 217 insertions(+))
TF Conversion: [Diff](https://github.com/modular-magician/terraform-google-conversion/compare/auto-pr-4981-old..auto-pr-4981) ( 1 file changed, 80 insertions(+))

modular-magician commented:
I have triggered VCR tests in RECORDING mode for the following tests that failed during VCR: TestAccAssuredWorkloadsWorkload_basic|TestAccAssuredWorkloadsWorkload_full|TestAccBigqueryDataTransferConfig|TestAccBigqueryReservationReservation_bigqueryReservationBasicExample|TestAccBillingBudget_budgetFilterProjectsOrdering|TestAccBillingBudget_billingBudgetUpdate|TestAccCloudBuildTrigger_pubsub_config|TestAccCloudBuildTrigger_webhook_config|TestAccComposerEnvironment_withMaintenanceWindow|TestAccComputeBackendService_withBackend|TestAccComputeBackendService_withBackendAndMaxUtilization|TestAccComputeBackendService_withBackendAndIAP|TestAccComputeBackendService_withMaxConnections|TestAccComputeBackendService_withMaxConnectionsPerInstance|TestAccComputeBackendService_withMaxRatePerEndpoint|TestAccComputeBackendService_withMaxConnectionsPerEndpoint|TestAccComputeGlobalForwardingRule_globalForwardingRuleInternalExample|TestAccComputeGlobalForwardingRule_internalLoadBalancing|TestAccComputeHaVpnGateway_computeHaVpnGatewayEncryptedInterconnectExample|TestAccComputeInstanceGroup_rename|TestAccComputeRouter_computeRouterEncryptedInterconnectExample|TestAccContainerCluster_backend|TestAccContainerNodePool_withGPU|TestAccDataLossPreventionDeidentifyTemplate_dlpDeidentifyTemplateBasicExample|TestAccDataLossPreventionDeidentifyTemplate_dlpDeidentifyTemplateUpdate|TestAccDialogflowCXFlow_dialogflowcxFlowFullExample|TestAccDialogflowCXVersion_dialogflowcxVersionFullExample|TestAccDialogflowCXFlow_update|TestAccDialogflowCXVersion_update|TestAccGKEHubMembership_gkehubMembershipBasicExample|TestAccGKEHubMembership_gkehubMembershipIssuerExample|TestAccNetworkServicesEdgeCacheKeyset_networkServicesEdgeCacheKeysetBasicExample|TestAccNetworkServicesEdgeCacheOrigin_networkServicesEdgeCacheOriginBasicExample|TestAccNetworkServicesEdgeCacheOrigin_networkServicesEdgeCacheOriginAdvancedExample|TestAccNetworkServicesEdgeCacheOrigin_updateAndImport|TestAccNetworkServicesEdgeCacheService_networkServicesEdgeCacheServiceBasicExample|TestAccNetworkServicesEdgeCacheService_networkServicesEdgeCacheServiceAdvancedExample|TestAccNetworkServicesEdgeCacheService_updateAndImport|TestAccPrivatecaCertificate_privatecaCertificateCsrExample|TestAccPrivatecaCertificateAuthority_privatecaCertificateAuthorityBasicExample|TestAccPrivatecaCertificate_privatecaCertificateConfigExample|TestAccPrivatecaCertificate_privatecaCertificateNoAuthorityExample|TestAccSecurityCenterNotificationConfig_sccNotificationConfigBasicExample You can view the result here: https://ci-oss.hashicorp.engineering/viewQueued.html?itemId=197963

modular-magician commented:
Tests failed during RECORDING mode: TestAccDataLossPreventionDeidentifyTemplate_dlpDeidentifyTemplateBasicExample|TestAccGKEHubMembership_gkehubMembershipBasicExample|TestAccPrivatecaCertificate_privatecaCertificateNoAuthorityExample|TestAccGKEHubMembership_gkehubMembershipIssuerExample Please fix these to complete your PR

melinath reviewed (dismissed):
Per discussion - this feature will need to be supported in both GA and beta; we don't support GA-only features. Marking as a change request to remove it from my review queue.

melinath commented:
@googlebot I fixed it.

modular-magician commented:
Hi!  I'm the modular magician.  Your PR generated some diffs in downstreams - here they are.
# Diff report:


Terraform GA: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-4981-old..auto-pr-4981) ( 3 files changed, 233 insertions(+), 2 deletions(-))
Terraform Beta: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-4981-old..auto-pr-4981) ( 5 files changed, 235 insertions(+), 5 deletions(-))
TF Conversion: [Diff](https://github.com/modular-magician/terraform-google-conversion/compare/auto-pr-4981-old..auto-pr-4981) ( 1 file changed, 80 insertions(+))

modular-magician commented:
I have triggered VCR tests in RECORDING mode for the following tests that failed during VCR: TestAccDataLossPreventionDeidentifyTemplate_dlpDeidentifyTemplateBasicExample|TestAccNotebooksInstance_notebookInstanceFullExample You can view the result here: https://ci-oss.hashicorp.engineering/viewQueued.html?itemId=199974

modular-magician commented:
Tests failed during RECORDING mode: TestAccNotebooksInstance_notebookInstanceFullExample Please fix these to complete your PR

rileykarson commented on mmv1/products/notebooks/api.yaml:
```diff
@@ -281,6 +281,42 @@ objects:
               Defines whether the instance has the vTPM enabled.
               Enabled by default.
             default_value: true
+      - !ruby/object:Api::Type::NestedObject
```
This should be a top-level enum called `nicType`, it isn't nested as far as I can tell.

rileykarson commented on mmv1/products/notebooks/api.yaml:
```diff
@@ -281,6 +281,42 @@ objects:
+              - VIRTIO_NET
+              - GVNIC          
+      - !ruby/object:Api::Type::NestedObject
+        name: 'ReservationAffinity'
```
```suggestion
        name: 'reservationAffinity'
```

rileykarson commented on mmv1/products/notebooks/api.yaml:
```diff
@@ -281,6 +281,42 @@ objects:
+          Reservation Affinity for consuming Zonal reservation.
+        properties:
+          - !ruby/object:Api::Type::Enum
+            name: 'type'
```
```suggestion
            name: 'type'
            required: true
```

rileykarson commented on mmv1/products/notebooks/api.yaml:
```diff
@@ -281,6 +281,42 @@ objects:
+            description: |
+              The type of Compute Reservation.
+            values:
+              - TYPE_UNSPECIFIED
```
```suggestion
```

X_UNSPECIFIED is always a dummy value, coming from proto -> rest translation

rileykarson commented on mmv1/templates/terraform/examples/notebook_instance_full.tf.erb:
```diff
@@ -28,6 +28,14 @@ resource "google_notebooks_instance" "<%= ctx[:primary_resource_id] %>" {
     terraform = "true"
   }
+
+  nic_type {
+    type = "VIRTIO_NET"
+  }
```
```suggestion
  nic_type = "VIRTIO_NET"
```

rileykarson reviewed (changes requested):
Seeing some errors, possible corrections inline:

```
Error: Error creating Instance: googleapi: Error 400: Invalid JSON payload received. Unknown name "NicType" at 'instance': Cannot find field.
Invalid JSON payload received. Unknown name "ReservationAffinity" at 'instance': Cannot find field.
Details:
[
{
"@type": "type.googleapis.com/google.rpc.BadRequest",
"fieldViolations": [
{
"description": "Invalid JSON payload received. Unknown name \"NicType\" at 'instance': Cannot find field.",
"field": "instance"
},
{
"description": "Invalid JSON payload received. Unknown name \"ReservationAffinity\" at 'instance': Cannot find field.",
"field": "instance"
}
]
}
]
on terraform_plugin_test.tf line 2, in resource "google_notebooks_instance" "instance":
2: resource "google_notebooks_instance" "instance" {
--- FAIL: TestAccNotebooksInstance_notebookInstanceFullExample (3.70s)
```

modular-magician commented:
Hi!  I'm the modular magician.  Your PR generated some diffs in downstreams - here they are.
# Diff report:


Terraform GA: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-4981-old..auto-pr-4981) ( 3 files changed, 229 insertions(+), 2 deletions(-))
Terraform Beta: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-4981-old..auto-pr-4981) ( 4 files changed, 230 insertions(+), 3 deletions(-))
TF Conversion: [Diff](https://github.com/modular-magician/terraform-google-conversion/compare/auto-pr-4981-old..auto-pr-4981) ( 1 file changed, 80 insertions(+))

modular-magician commented:
Hi!  I'm the modular magician.  Your PR generated some diffs in downstreams - here they are.
# Diff report:


Terraform GA: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-4981-old..auto-pr-4981) ( 3 files changed, 229 insertions(+), 2 deletions(-))
Terraform Beta: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-4981-old..auto-pr-4981) ( 4 files changed, 230 insertions(+), 3 deletions(-))
TF Conversion: [Diff](https://github.com/modular-magician/terraform-google-conversion/compare/auto-pr-4981-old..auto-pr-4981) ( 1 file changed, 80 insertions(+))

modular-magician commented:
Error trying to cancel build ()

modular-magician commented:
Error trying to cancel build ()

modular-magician commented:
Error trying to cancel build ()

modular-magician commented:
Error trying to cancel build ()

modular-magician commented:
Error trying to cancel build ()

modular-magician commented:
Error trying to cancel build ()

modular-magician commented:
Error trying to cancel build ()

modular-magician commented:
Error trying to cancel build ()

modular-magician commented:
Hi!  I'm the modular magician.  Your PR generated some diffs in downstreams - here they are.
# Diff report:


Terraform GA: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-4981-old..auto-pr-4981) ( 3 files changed, 229 insertions(+), 2 deletions(-))
Terraform Beta: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-4981-old..auto-pr-4981) ( 4 files changed, 230 insertions(+), 3 deletions(-))
TF Conversion: [Diff](https://github.com/modular-magician/terraform-google-conversion/compare/auto-pr-4981-old..auto-pr-4981) ( 1 file changed, 80 insertions(+))

modular-magician commented:
Error trying to cancel build ()

modular-magician commented:
Error trying to cancel build ()

modular-magician commented:
Error trying to cancel build ()

modular-magician commented:
Error trying to cancel build ()

rileykarson commented on mmv1/products/notebooks/api.yaml:
```diff
@@ -281,6 +281,42 @@ objects:
+        name: 'nicType'
+        description: |
+          The type of vNIC driver.
+        properties:
```
Strings can't have subproperties, you'll want to remove these.

modular-magician commented:
Hi!  I'm the modular magician.  Your PR generated some diffs in downstreams - here they are.
# Diff report:


Terraform GA: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-4981-old..auto-pr-4981) ( 3 files changed, 179 insertions(+), 2 deletions(-))
Terraform Beta: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-4981-old..auto-pr-4981) ( 4 files changed, 180 insertions(+), 3 deletions(-))
TF Conversion: [Diff](https://github.com/modular-magician/terraform-google-conversion/compare/auto-pr-4981-old..auto-pr-4981) ( 1 file changed, 61 insertions(+))

modular-magician commented:
Error trying to cancel build ()

modular-magician commented:
Error trying to cancel build ()

modular-magician commented:
Error trying to cancel build ()

modular-magician commented:
Error trying to cancel build ()

modular-magician commented:
Error trying to cancel build ()

rileykarson commented on mmv1/products/notebooks/api.yaml:
```diff
@@ -281,6 +281,37 @@ objects:
+        description: |
+          The type of vNIC driver.
+        values:
+          - UNSPECIFIED_NIC_TYPE
```
```suggestion
```

gogasca replied:
This is a default that is not standard. UNSPECIFIED_NIC_TYPE vs NIC_TYPE_UNSPECIFIED. This is because we use Compute API under the hood and decided to use same default value. https://cloud.google.com/workflows/docs/reference/googleapis/compute/v1/Overview (Search for UNSPECIFIED_NIC_TYPE)

rileykarson commented:
Waiting on [VCR] test results.

gogasca commented:
> It looks like `TestAccNotebooksInstance_notebookInstanceFullExample` is currently failing in VCR with:
> 
> ```
> Error: Error creating Instance: Post "https://notebooks.googleapis.com/v1beta1/projects/<project-id>/locations/<location>/instances?alt=json&instanceId=tf-test-notebooks-instancecjyyl6xydx": Requested interaction not found
> ```
> 
> It looks like this is from the REPLAYING run & indicates a missing cassette; the RECORDING run is never triggered. i.e. it's a CI issue.

Sounds good, just replied to few comments now that Github is back.

modular-magician commented:
Hi!  I'm the modular magician.  Your PR generated some diffs in downstreams - here they are.
# Diff report:


Terraform GA: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-4981-old..auto-pr-4981) ( 3 files changed, 177 insertions(+))
Terraform Beta: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-4981-old..auto-pr-4981) ( 3 files changed, 177 insertions(+))
TF Conversion: [Diff](https://github.com/modular-magician/terraform-google-conversion/compare/auto-pr-4981-old..auto-pr-4981) ( 1 file changed, 61 insertions(+))

modular-magician commented:
Error trying to cancel build ()

modular-magician commented:
Error trying to cancel build ()

modular-magician commented:
Error trying to cancel build ()

modular-magician commented:
Error trying to cancel build ()

modular-magician commented:
Error trying to cancel build ()

modular-magician commented:
Error trying to cancel build ()

modular-magician commented:
Error trying to cancel build ()

melinath commented:
/gcbrun - CI issue is now fixed

modular-magician commented:
Hi!  I'm the modular magician.  Your PR generated some diffs in downstreams - here they are.
# Diff report:


Terraform GA: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-4981-old..auto-pr-4981) ( 3 files changed, 177 insertions(+))
Terraform Beta: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-4981-old..auto-pr-4981) ( 3 files changed, 177 insertions(+))
TF Conversion: [Diff](https://github.com/modular-magician/terraform-google-conversion/compare/auto-pr-4981-old..auto-pr-4981) ( 1 file changed, 61 insertions(+))

modular-magician commented:
Error trying to cancel build ()

modular-magician commented:
Error trying to cancel build ()

modular-magician commented:
Error trying to cancel build ()

modular-magician commented:
Error trying to cancel build ()

modular-magician commented:
Error trying to cancel build ()

modular-magician commented:
Error trying to cancel build ()

modular-magician commented:
Error trying to cancel build ()

melinath commented:
It looks like this ended up being queued for too long & didn't re-run the specific tests.

Here's a non-VCR build with just the full example test: https://ci-oss.hashicorp.engineering/buildConfiguration/GoogleCloudBeta_ProviderGoogleCloudBetaMmUpstream/200665

gogasca commented:
> It looks like this ended up being queued for too long & didn't re-run the specific tests.
> 
> Here's a non-VCR build with just the full example test: https://ci-oss.hashicorp.engineering/buildConfiguration/GoogleCloudBeta_ProviderGoogleCloudBetaMmUpstream/200665

Not sure what I need to do here?

rileykarson commented:
No action for you @gogasca

It turns out a typo in CI config made the test get skipped, I fixed that for the build and kicked it back off. Results will take an hour or two.

rileykarson commented on mmv1/products/notebooks/api.yaml:
```diff
@@ -281,6 +281,37 @@ objects:
+        description: |
+          Reservation Affinity for consuming Zonal reservation.
+        properties:
+          - !ruby/object:Api::Type::Enum
```
I can't comment directly on L298 for some reason. Weird. Are you sure the field is called `type`? While the type of the field is called `Type`, the field itself is called `consume_reservation_type` in the RPC docs: https://cloud.google.com/notebooks/docs/reference/rpc/google.cloud.notebooks.v1#google.cloud.notebooks.v1.ReservationAffinity

gogasca replied:
Good catch, changing this now.
Github has been having all sorts of issues since yesterday.

rileykarson commented on mmv1/products/notebooks/api.yaml:
```diff
@@ -281,6 +281,37 @@ objects:
               Enabled by default.
             default_value: true
+      - !ruby/object:Api::Type::Enum
+        name: 'nicType'
```
Given that this feature will be in public preview please add `min_version: beta` to each new field, and register a beta version at the top of this file. It should point to the REST API that this feature will be exposed under.

gogasca replied:
ack

rileykarson reviewed (commented):
Well, we get an error back if the test fails. Resolution inline.

```
------- Stdout: -------
=== RUN   TestAccNotebooksInstance_notebookInstanceFullExample
=== PAUSE TestAccNotebooksInstance_notebookInstanceFullExample
=== CONT  TestAccNotebooksInstance_notebookInstanceFullExample
provider_test.go:276: Step 1/2 error: Error running apply: exit status 1
2021/08/11 17:08:45 [DEBUG] Using modified User-Agent: Terraform/0.12.30 HashiCorp-terraform-exec/0.13.0
Error: Error creating Instance: googleapi: Error 400: Invalid JSON payload received. Unknown name "type" at 'instance.reservation_affinity': Cannot find field.
Details:
[
{
"@type": "type.googleapis.com/google.rpc.BadRequest",
"fieldViolations": [
{
"description": "Invalid JSON payload received. Unknown name \"type\" at 'instance.reservation_affinity': Cannot find field.",
"field": "instance.reservation_affinity"
}
]
}
]
on terraform_plugin_test.tf line 2, in resource "google_notebooks_instance" "instance":
2: resource "google_notebooks_instance" "instance" {
--- FAIL: TestAccNotebooksInstance_notebookInstanceFullExample (2.70s)
FAIL
```

modular-magician commented:
Hi!  I'm the modular magician.  Your PR generated some diffs in downstreams - here they are.
# Diff report:


Terraform GA: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-4981-old..auto-pr-4981) ( 2 files changed, 37 insertions(+))
Terraform Beta: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-4981-old..auto-pr-4981) ( 4 files changed, 178 insertions(+), 1 deletion(-))

modular-magician commented:
Error trying to cancel build ()

modular-magician commented:
Error trying to cancel build ()

modular-magician commented:
Error trying to cancel build ()

modular-magician commented:
Error trying to cancel build ()

modular-magician commented:
Error trying to cancel build ()

modular-magician commented:
Error trying to cancel build ()

modular-magician commented:
Error trying to cancel build ()

modular-magician commented:
Error trying to cancel build ()

modular-magician commented:
Hi!  I'm the modular magician.  Your PR generated some diffs in downstreams - here they are.
# Diff report:


Terraform GA: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-4981-old..auto-pr-4981) ( 2 files changed, 37 insertions(+))
Terraform Beta: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-4981-old..auto-pr-4981) ( 4 files changed, 178 insertions(+), 1 deletion(-))

modular-magician commented:
Error trying to cancel build ()

modular-magician commented:
Error trying to cancel build ()

modular-magician commented:
Error trying to cancel build ()

modular-magician commented:
Error trying to cancel build ()

modular-magician commented:
Error trying to cancel build ()

modular-magician commented:
Error trying to cancel build ()

modular-magician commented:
Error trying to cancel build ()

modular-magician commented:
Error trying to cancel build ()

rileykarson commented:
The API is accepting the `nic_type` and `reservation_affinity` values, but is not returning them. Moving forward is blocked until they are returned.

```
------- Stdout: -------
=== RUN   TestAccNotebooksInstance_notebookInstanceFullExample
=== PAUSE TestAccNotebooksInstance_notebookInstanceFullExample
=== CONT  TestAccNotebooksInstance_notebookInstanceFullExample
provider_test.go:276: Step 1/2 error: After applying this test step, the plan was not empty.
stdout:
An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
-/+ destroy and then create replacement
Terraform will perform the following actions:
# google_notebooks_instance.instance must be replaced
-/+ resource "google_notebooks_instance" "instance" {
boot_disk_size_gb  = 110
boot_disk_type     = "PD_SSD"
~ create_time        = "2021-08-11T18:37:20.571975308Z" -> (known after apply)
~ id                 = "projects/ci-test-project-188019/locations/us-central1-a/instances/tf-test-notebooks-instance9stbsw1iss" -> (known after apply)
install_gpu_driver = true
instance_owners    = [
"admin@hashicorptest.com",
]
~ labels             = {
- "goog-caip-notebook" = "" -> null
"k"                  = "val"
}
location           = "us-central1-a"
machine_type       = "e2-medium"
metadata           = {
"terraform" = "true"
}
name               = "tf-test-notebooks-instance9stbsw1iss"
~ network            = "https://www.googleapis.com/compute/v1/projects/ci-test-project-188019/global/networks/default" -> "projects/ci-test-project-188019/global/networks/default"
+ nic_type           = "VIRTIO_NET"
no_proxy_access    = true
no_public_ip       = true
~ project            = "ci-test-project-188019" -> (known after apply)
+ proxy_uri          = (known after apply)
service_account    = "hashicorp-test-runner@ci-test-project-188019.iam.gserviceaccount.com"
~ state              = "PROVISIONING" -> (known after apply)
~ subnet             = "https://www.googleapis.com/compute/v1/projects/ci-test-project-188019/regions/us-central1/subnetworks/default" -> "projects/ci-test-project-188019/regions/us-central1/subnetworks/default"
~ update_time        = "2021-08-11T18:37:44.138406509Z" -> (known after apply)
+ reservation_affinity {
+ consume_reservation_type = "NO_RESERVATION"
}
+ shielded_instance_config {
+ enable_integrity_monitoring = (known after apply)
+ enable_secure_boot          = (known after apply)
+ enable_vtpm                 = (known after apply)
}
~ vm_image {
image_family = "tf-latest-cpu"
project      = "deeplearning-platform-release"
}
}
Plan: 1 to add, 0 to change, 1 to destroy.
--- FAIL: TestAccNotebooksInstance_notebookInstanceFullExample (67.67s)
FAIL
```

modular-magician commented:
I have triggered VCR tests in RECORDING mode for the following tests that failed during VCR: TestAccNotebooksInstance_notebookInstanceFullExample|TestAccPrivatecaCertificateAuthority_privatecaCertificateAuthoritySubordinateExample You can view the result here: https://ci-oss.hashicorp.engineering/viewQueued.html?itemId=200695

modular-magician commented:
I have triggered VCR tests in RECORDING mode for the following tests that failed during VCR: TestAccNotebooksInstance_notebookInstanceFullExample|TestAccPrivatecaCertificateAuthority_privatecaCertificateAuthoritySubordinateExample You can view the result here: https://ci-oss.hashicorp.engineering/viewQueued.html?itemId=200695

modular-magician commented:
Tests failed during RECORDING mode: TestAccNotebooksInstance_notebookInstanceFullExample Please fix these to complete your PR

modular-magician commented:
Tests failed during RECORDING mode: TestAccNotebooksInstance_notebookInstanceFullExample Please fix these to complete your PR

gogasca commented:
> The API is accepting the `nic_type` and `reservation_affinity` values, but is not returning them. Moving forward is blocked until they are returned.
> 
> ```
> ------- Stdout: -------
> === RUN   TestAccNotebooksInstance_notebookInstanceFullExample
> === PAUSE TestAccNotebooksInstance_notebookInstanceFullExample
> === CONT  TestAccNotebooksInstance_notebookInstanceFullExample
> provider_test.go:276: Step 1/2 error: After applying this test step, the plan was not empty.
> stdout:
> An execution plan has been generated and is shown below.
> Resource actions are indicated with the following symbols:
> -/+ destroy and then create replacement
> Terraform will perform the following actions:
> # google_notebooks_instance.instance must be replaced
> -/+ resource "google_notebooks_instance" "instance" {
> boot_disk_size_gb  = 110
> boot_disk_type     = "PD_SSD"
> ~ create_time        = "2021-08-11T18:37:20.571975308Z" -> (known after apply)
> ~ id                 = "projects/ci-test-project-188019/locations/us-central1-a/instances/tf-test-notebooks-instance9stbsw1iss" -> (known after apply)
> install_gpu_driver = true
> instance_owners    = [
> "admin@hashicorptest.com",
> ]
> ~ labels             = {
> - "goog-caip-notebook" = "" -> null
> "k"                  = "val"
> }
> location           = "us-central1-a"
> machine_type       = "e2-medium"
> metadata           = {
> "terraform" = "true"
> }
> name               = "tf-test-notebooks-instance9stbsw1iss"
> ~ network            = "https://www.googleapis.com/compute/v1/projects/ci-test-project-188019/global/networks/default" -> "projects/ci-test-project-188019/global/networks/default"
> + nic_type           = "VIRTIO_NET"
> no_proxy_access    = true
> no_public_ip       = true
> ~ project            = "ci-test-project-188019" -> (known after apply)
> + proxy_uri          = (known after apply)
> service_account    = "hashicorp-test-runner@ci-test-project-188019.iam.gserviceaccount.com"
> ~ state              = "PROVISIONING" -> (known after apply)
> ~ subnet             = "https://www.googleapis.com/compute/v1/projects/ci-test-project-188019/regions/us-central1/subnetworks/default" -> "projects/ci-test-project-188019/regions/us-central1/subnetworks/default"
> ~ update_time        = "2021-08-11T18:37:44.138406509Z" -> (known after apply)
> + reservation_affinity {
> + consume_reservation_type = "NO_RESERVATION"
> }
> + shielded_instance_config {
> + enable_integrity_monitoring = (known after apply)
> + enable_secure_boot          = (known after apply)
> + enable_vtpm                 = (known after apply)
> }
> ~ vm_image {
> image_family = "tf-latest-cpu"
> project      = "deeplearning-platform-release"
> }
> }
> Plan: 1 to add, 0 to change, 1 to destroy.
> --- FAIL: TestAccNotebooksInstance_notebookInstanceFullExample (67.67s)
> FAIL
> ```

Ack, adding this info in API. Will rollout ASAP and update

gogasca commented:
API returns now correct information:

```
{
  "name": "projects/gogasca-autopush-notebooks/locations/us-west1-a/operations/operation-1629325528306-5c9dce956950e-0d51ed84-3d4f87b5",
  "metadata": {
    "@type": "type.googleapis.com/google.cloud.notebooks.v1beta1.OperationMetadata",
    "createTime": "2021-08-18T22:25:28.528456286Z",
    "endTime": "2021-08-18T22:25:50.302220904Z",
    "target": "projects/gogasca-autopush-notebooks/locations/us-west1-a/instances/ref-3a",
    "verb": "create",
    "requestedCancellation": false,
    "apiVersion": "v1beta1",
    "endpoint": "CreateInstance"
  },
  "done": true,
  "response": {
    "@type": "type.googleapis.com/google.cloud.notebooks.v1beta1.Instance",
    "name": "projects/gogasca-autopush-notebooks/locations/us-west1-a/instances/ref-3a",
    "serviceAccount": "634850243685-compute@developer.gserviceaccount.com",
    "machineType": "https://www.googleapis.com/compute/v1/projects/gogasca-autopush-notebooks/zones/us-west1-a/machineTypes/n1-standard-1",
    "state": "PROVISIONING",
    "network": "https://www.googleapis.com/compute/v1/projects/gogasca-autopush-notebooks/global/networks/default",
    "subnet": "https://www.googleapis.com/compute/v1/projects/gogasca-autopush-notebooks/regions/us-west1/subnetworks/default",
    "labels": {
      "goog-caip-notebook": ""
    },
    "metadata": {
      "shutdown-script": "/opt/deeplearning/bin/shutdown_script.sh",
      "notebooks-api": "AUTOPUSH",
      "proxy-mode": "service_account",
      "proxy-user-mail": "gogasca@google.com",
      "enable-guest-attributes": "TRUE",
      "report-system-health": "TRUE"
    },
    "createTime": "2021-08-18T22:25:28.523521866Z",
    "updateTime": "2021-08-18T22:25:28.523521866Z",
    "nicType": "VIRTIO_NET",
    "reservationAffinity": {
      "consumeReservationType": "NO_RESERVATION"
    }
  }
}
```

rileykarson commented:
/gcbrun

gogasca commented:
Checking if this time we have better luck?

rileykarson commented:
Ah, looks like generation failed- can you rebase this PR? That should fix it.

gogasca commented:
Made a small change.

rileykarson commented:
I specifically need you to rebase so that the build definitions under the `.ci/` folder are updated.

modular-magician commented:
Hi!  I'm the modular magician.  Your PR generated some diffs in downstreams - here they are.
# Diff report:


Terraform GA: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-4981-old..auto-pr-4981) ( 2 files changed, 37 insertions(+))
Terraform Beta: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-4981-old..auto-pr-4981) ( 4 files changed, 178 insertions(+), 1 deletion(-))

modular-magician commented:
Error trying to cancel build ()

modular-magician commented:
Error trying to cancel build ()

modular-magician commented:
Error trying to cancel build ()

modular-magician commented:
Error trying to cancel build ()

modular-magician commented:
Error trying to cancel build ()

modular-magician commented:
Error trying to cancel build ()

modular-magician commented:
Error trying to cancel build ()

modular-magician commented:
Error trying to cancel build ()

modular-magician commented:
Error trying to cancel build ()

modular-magician commented:
Error trying to cancel build ()

gogasca commented:
> I specifically need you to rebase so that the build definitions under the `.ci/` folder are updated.

Done

rileykarson commented:
/gcbrun

modular-magician commented:
Hi!  I'm the modular magician.  Your PR generated some diffs in downstreams - here they are.
# Diff report:


Terraform GA: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-4981-old..auto-pr-4981) ( 2 files changed, 37 insertions(+))
Terraform Beta: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-4981-old..auto-pr-4981) ( 6 files changed, 181 insertions(+), 2 deletions(-))

modular-magician commented:
Error trying to cancel build ()

modular-magician commented:
Error trying to cancel build ()

modular-magician commented:
Error trying to cancel build ()

modular-magician commented:
Error trying to cancel build ()

modular-magician commented:
Error trying to cancel build ()

modular-magician commented:
Error trying to cancel build ()

modular-magician commented:
Error trying to cancel build ()

modular-magician commented:
Error trying to cancel build ()

modular-magician commented:
Error trying to cancel build ()

modular-magician commented:
Error trying to cancel build ()

modular-magician commented:
Error trying to cancel build ()

modular-magician commented:
I have triggered VCR tests in RECORDING mode for the following tests that failed during VCR:  You can view the result here: https://ci-oss.hashicorp.engineering/viewQueued.html?itemId=201959

modular-magician commented:
Tests failed during RECORDING mode: 2021/08/19 21:31:40 suppressing diff "push_config.0.attributes.x-goog-version" with old "", new "v1"|2021/08/19 21:31:40 suppressing diff "push_config.0.attributes.x-goog-version" with old "v1", new "v2"|2021/08/19 21:31:40 suppressing diff "push_config.0.attributes.x-goog-version" with old "v1", new "v1"|2021/08/19 21:31:43 days: 0 0|2021/08/19 21:31:43 days: 3 2|2021/08/19 21:31:43 duration:  72h|2021/08/19 21:31:43 days: 3 0|2021/08/19 21:31:43 duration:|2021/08/19 21:31:43 duration: 3h 4h|2021/08/19 21:31:43 duration:  70h|runtime.mapassign_faststr(0x3157fa0, 0xc0002808a0, 0xc00039a040, 0x1c, 0x4ffaa00)|runtime.throw(0x365474b, 0x15)|fatal error: concurrent map writes|github.com/hashicorp/terraform-provider-google-beta/google-beta.TestAccTags.func1(0xc00011e000)|github.com/hashicorp/terraform-provider-google-beta/google-beta.vcrSource(0xc00011e000, 0xc000052109, 0x73, 0xc0000500e1, 0x9, 0xc000164c60, 0x49a286, 0x611ecdbf)|github.com/hashicorp/terraform-provider-google-beta/google-beta.testAccTagsTagKeyIamBinding(0xc00011e000)|testing.tRunner(0xc00011e000, 0xc0017e0240)|github.com/hashicorp/terraform-provider-google-beta/google-beta.randString(0xc00011e000, 0xa, 0x0, 0x439)|created by testing.(*T).Run|runtime.goexit|testing.(*T).Run(0xc00064a900, 0x36378c6, 0xb, 0x3958980, 0x49a201)|testing.runTests.func1(0xc00052e000)|testing.tRunner(0xc00052e000, 0xc0003bbd20) Please fix these to complete your PR

rileykarson commented:
Man, CI is not happy w/ me right now. Gonna try again.

/gcbrun

modular-magician commented:
Hi!  I'm the modular magician.  Your PR generated some diffs in downstreams - here they are.
# Diff report:


Terraform GA: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-4981-old..auto-pr-4981) ( 2 files changed, 37 insertions(+))
Terraform Beta: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-4981-old..auto-pr-4981) ( 3 files changed, 177 insertions(+))

modular-magician commented:
Error trying to cancel build ()

modular-magician commented:
Error trying to cancel build ()

modular-magician commented:
Error trying to cancel build ()

modular-magician commented:
Error trying to cancel build ()

modular-magician commented:
Error trying to cancel build ()

modular-magician commented:
Error trying to cancel build ()

modular-magician commented:
Error trying to cancel build ()

modular-magician commented:
Error trying to cancel build ()

modular-magician commented:
Error trying to cancel build ()

modular-magician commented:
Error trying to cancel build ()

modular-magician commented:
Error trying to cancel build ()

modular-magician commented:
Error trying to cancel build ()

rileykarson commented:
Over-aggressive `Fixes: ` message, GitHub didn't indicate it would do that!

ScottSuarez commented:
> Over-aggressive Fixes: message, GitHub didn't indicate it would do that!

yeah i don't know why they got rid of the squiggly thing

ScottSuarez commented:
oh wait, an closing an issue can close prs.... weird

modular-magician commented:
Hi!  I'm the modular magician.  Your PR generated some diffs in downstreams - here they are.
# Diff report:


Terraform GA: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-4981-old..auto-pr-4981) ( 2 files changed, 37 insertions(+))
Terraform Beta: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-4981-old..auto-pr-4981) ( 5 files changed, 179 insertions(+), 2 deletions(-))

modular-magician commented:
Error trying to cancel build ()

modular-magician commented:
Error trying to cancel build ()

modular-magician commented:
Error trying to cancel build ()

modular-magician commented:
Error trying to cancel build ()

modular-magician commented:
Error trying to cancel build ()

modular-magician commented:
Error trying to cancel build ()

modular-magician commented:
Error trying to cancel build ()

modular-magician commented:
Error trying to cancel build ()

modular-magician commented:
Error trying to cancel build ()

modular-magician commented:
Error trying to cancel build ()

modular-magician commented:
Error trying to cancel build ()

modular-magician commented:
Error trying to cancel build ()

gogasca commented:
I assume at this point I just need to wait for another test?

modular-magician commented:
I have triggered VCR tests in RECORDING mode for the following tests that failed during VCR: TestAccComposerEnvironment_update|TestAccComposerEnvironment_withSoftwareConfig|TestAccNotebooksInstance_notebookInstanceFullExample|TestAccPrivatecaCaPool_privatecaCapoolEmptyBaseline|TestAccPrivatecaCertificateAuthority_privatecaCertificateAuthoritySubordinateExample You can view the result here: https://ci-oss.hashicorp.engineering/viewQueued.html?itemId=202080

rileykarson commented on mmv1/products/notebooks/api.yaml:
```diff
@@ -281,6 +281,39 @@ objects:
+          Reservation Affinity for consuming Zonal reservation.
+        properties:
+          - !ruby/object:Api::Type::Enum
+            name: 'consume_reservation_type'
```
```suggestion
            name: 'consumeReservationType'
```

This caused us to send the message

```
 "reservationAffinity": {
  "consume_reservation_type": "NO_RESERVATION"
 },
```

instead of

```
 "reservationAffinity": {
  "consumeReservationType": "NO_RESERVATION"
 },
```

and then look for the key `consume_reservation_type` instead of `consumeReservationType` in the response.

modular-magician commented:
Hi!  I'm the modular magician.  Your PR generated some diffs in downstreams - here they are.
# Diff report:


Terraform GA: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-4981-old..auto-pr-4981) ( 2 files changed, 37 insertions(+))
Terraform Beta: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-4981-old..auto-pr-4981) ( 4 files changed, 178 insertions(+), 1 deletion(-))

modular-magician commented:
Error trying to cancel build ()

modular-magician commented:
Error trying to cancel build ()

modular-magician commented:
Error trying to cancel build ()

modular-magician commented:
Error trying to cancel build ()

modular-magician commented:
Error trying to cancel build ()

modular-magician commented:
Error trying to cancel build ()

modular-magician commented:
Error trying to cancel build ()

modular-magician commented:
Error trying to cancel build ()

modular-magician commented:
Error trying to cancel build ()

modular-magician commented:
Error trying to cancel build ()

modular-magician commented:
Error trying to cancel build ()

modular-magician commented:
Error trying to cancel build ()

modular-magician commented:
Error trying to cancel build ()

modular-magician commented:
I have triggered VCR tests in RECORDING mode for the following tests that failed during VCR: TestAccComposerEnvironment_update|TestAccComposerEnvironment_withSoftwareConfig|TestAccNotebooksInstance_notebookInstanceFullExample|TestAccPrivatecaCaPool_privatecaCapoolUpdate|TestAccPrivatecaCaPool_privatecaCapoolAllFieldsExample|TestAccPrivatecaCaPool_privatecaCapoolEmptyBaseline You can view the result here: https://ci-oss.hashicorp.engineering/viewQueued.html?itemId=202154

rileykarson commented:
@melinath can you dismiss your change request? Those are set to block merges.

modular-magician commented:
Tests failed during RECORDING mode: TestAccPrivatecaCaPool_privatecaCapoolEmptyBaseline|TestAccComposerEnvironment_withSoftwareConfig|TestAccComposerEnvironment_update Please fix these to complete your PR

melinath commented:
@rileykarson done
